### ISSUE TITLE
Sunfury Arcane Mage opener#2

### ISSUE THREAD

**Issue by GitBearded** [Hekili/hekili#3654]
### Before You Begin

- [X] I confirm that I have downloaded the latest version of the addon.
- [X] I am not playing on a private server.
- [X] I checked for an [existing, open ticket](https://github.com/Hekili/hekili/labels/recommendation) for this issue and was not able to find one.
- [X] I edited the title of this issue (above) so that it describes the issue I am reporting.
- [X] I am reporting an issue with the default priority included with the specialization (imported or edited priorities are not supported).

### Describe the Issue

After Arcane Surge, touch of the magi into arcane blast should be the recommended order or spells, if CC procs on first blast it goes into missiles but then back into arcane blast, until 6 blasts have been cast to proc burden of power, on the last arcane blast cast (7th) it should recommend barrage to instantly consume Glorious Incandescence. 

### How to Reproduce

Do the rotation.

### Snapshot (Link)

https://pastebin.com/kPKnTpYM

### Raidbots Sim Report (Link)

https://www.raidbots.com/simbot/report/nw4D21ixQRH9WYS7KHMXaH

### Contact Information

COLOSSAL

**Comment by Hekili** [comment]
In this case, Touch of the Magi had 0.45 seconds left on its cooldown.

```
        26.  touch_of_the_magi ( cd_opener - 1 )
        The action (touch_of_the_magi) is usable at (2.13 + 0.45) with cost of 137500 mana (have 139453).
         - the action is ready before the current recommendation (at +0.45 vs. +10.00).
        List ( cd_opener ) called from ( Arcane:default:16 ) would PASS at 0.45.
        NONE
         - this entry's criteria PASSES: prev_gcd[1].arcane_barrage[false] & ( action.arcane_barrage.in_flight_remains[0.00] <= 0.5 | gcd.remains[0.00] <= 0.5 ) & ( buff.arcane_surge.up[true] | cooldown.arcane_surge.remains[89.55] > 30 ) | ( prev_gcd[1].arcane_surge[true] & buff.arcane_charge.stack[1.00] < 4 )
        Action chosen:  touch_of_the_magi at 0.45!
```

That led the addon to continue down the action list, selecting Arcane Orb which was available 0.45s sooner.  However, the list that Arcane Orb was selected from will no longer be used/available in the next release.

In the meantime, you could consider setting the "Clash" value of the Touch of the Magi ability a bit higher; the addon will treat it as available a little sooner than it actually is, to avoid moving further down the action list.

**Comment by syrifgit** [comment]
> In this case, Touch of the Magi had 0.45 seconds left on its cooldown.
> 
> ```
>         26.  touch_of_the_magi ( cd_opener - 1 )
>         The action (touch_of_the_magi) is usable at (2.13 + 0.45) with cost of 137500 mana (have 139453).
>          - the action is ready before the current recommendation (at +0.45 vs. +10.00).
>         List ( cd_opener ) called from ( Arcane:default:16 ) would PASS at 0.45.
>         NONE
>          - this entry's criteria PASSES: prev_gcd[1].arcane_barrage[false] & ( action.arcane_barrage.in_flight_remains[0.00] <= 0.5 | gcd.remains[0.00] <= 0.5 ) & ( buff.arcane_surge.up[true] | cooldown.arcane_surge.remains[89.55] > 30 ) | ( prev_gcd[1].arcane_surge[true] & buff.arcane_charge.stack[1.00] < 4 )
>         Action chosen:  touch_of_the_magi at 0.45!
> ```
> 
> That led the addon to continue down the action list, selecting Arcane Orb which was available 0.45s sooner. However, the list that Arcane Orb was selected from will no longer be used/available in the next release.
> 
> In the meantime, you could consider setting the "Clash" value of the Touch of the Magi ability a bit higher; the addon will treat it as available a little sooner than it actually is, to avoid moving further down the action list.

It wasn't actually the spell cooldown, it was the lack of mana / mana regen. The 0.45 window was how long it thought it would take to regen the mana, but it actually regens enough to cast it basically instantly. Reducing the mana cost to 0 allows the opener to proceed as it does in the sims and guides (See my mana regen question on discord)

Clash also did fix it as a different solution too.
